python/perf-kernels/flash-attention.py

```diff
@@ -207,14 +206,14 @@ def _attn_fwd_inner(
         # For padded blocks, we will overrun the tensor size if
         # we load all BLOCK_N. For others, the blocks are all within range.
         if PADDED_BLOCK:
-            k = tl.load(K_block_ptr, boundary_check=(1,), padding_option="zero")
+            k = tl.load(K_block_ptr, boundary_check=(1,0), padding_option="zero")
         else:
-            k = tl.load(K_block_ptr)
+            k = tl.load(K_block_ptr, boundary_check=(0,), padding_option="zero")
```

Comment:
This is going to cause performance to tank. Can you do the boundary check optionally when head size is not a power of 2?

Reply:
fixed